lucene/core/src/java/org/apache/lucene/search/ConstantScoreScorer.java

```diff
@@ -70,7 +120,7 @@ public TwoPhaseIterator twoPhaseIterator() {
 
   @Override
   public int docID() {
-    return disi.docID();
```

Comment:
You can keep `disi.docID()` here and move `doc` to the `DocIdSetIteratorWrapper` ?

Reply:
Totally, fixed in 114013f86a7a3b12aa70e7ee6bea4d1211a1448b, thanks :+1: